test-data/unit/check-classes.test

```diff
@@ -6178,3 +6178,15 @@ class B(A):
     def meth(cls: Type[T]) -> List[T]: ...
 
 [builtins fixtures/isinstancelist.pyi]
+
+[case testCheckUntypedDefsSelf]
+# flags: --check-untyped-defs
+class Foo:
+    def foo(self):
+        self.bar()
+        self.baz()  # E: "Foo" has no attribute "baz"
+
+    @classmethod
+    def bar(cls):
+        cls.baz()  # E: "Type[Foo]" has no attribute "baz"
+[builtins fixtures/classmethod.pyi]
```

Comment:
I would add also tests for two other reported problems:
* An attribute initialized with `0` in untyped `__init__` has revealed type `int` in another untyped method
* Descriptors also work as expected in untyped methods.

Reply:
Also, if an attribute is initialized to `None` in `__init__` the inferred type should probably be either `Optional[Any]` or `Any` (the first one may generate many false positives). If an attribute is initialized to `[]`, the inferred type should probably be `List[Any]`, and I think that we shouldn't require a type annotation.